globals.go

```diff
@@ -94,16 +104,13 @@ func (r *rawGlobals) eval(meta StackMetadata) (*Globals, error) {
 		return nil, err
 	}
 
+	globals := newGlobals()
 	// error messages improve if globals is empty instead of undefined
-	empty, err := hclMapToCty(nil)
-	if err != nil {
-		return nil, fmt.Errorf("initializing empty globals: %v", err)
+	if err := globals.addToEvalCtx(evalctx); err != nil {
```

Comment:
now the global is not empty before the evaluations but contains meta vars in it. Doesn't this change the behavior if no variable gets evaluated?

Reply:
globals is empty, the code is basically:

```
globals := newGlobals()
if err := globals.addToEvalCtx(evalctx); err != nil {
}
```

This means:

* Create new/empty globals
* Add empty globals to evalctx

**newGlobals** is just this:

```go
func newGlobals() *Globals {
	return &Globals{
		attributes: map[string]cty.Value{},
	}
}
```